core/src/services/webhdfs/writer.rs

```diff
@@ -64,6 +58,69 @@ impl oio::OneShotWrite for WebhdfsWriter {
             _ => Err(parse_error(resp).await?),
         }
     }
+
+    async fn write_block(&self, size: u64, block_id: String, body: AsyncBody) -> Result<()> {
+        let req = self
+            .backend
+            .webhdfs_create_object_request(&block_id, Some(size), &self.op, body)
```

Comment:
Store temporary files in `{tmp_dir}/{block_id}` to prevent duplication and ensure they don't interfere with the user list results. It can be annoying and unexpected for users to encounter a list of temporary files when using the `list` command.

What the error while trying to concat to path that not in the same dir?

Reply:
```json
{
    "RemoteException": {
        "exception": "HadoopIllegalArgumentException",
        "javaClassName": "org.apache.hadoop.HadoopIllegalArgumentException",
        "message": "Source file /test/path.txt is not in the same directory with the target /path.txt"
    }
}
```

Reply:
Ok, I got it. So we need to implement in this way:

- We should add `atomic_write_dir` like we do for `hdfs` and `fs`
- We should write and concat into `atomic_write_dir` first.
- Rename the file to target path

Reply:
There's a slightly cumbersome method where we can concatenate files in the tmp folder and then rename them to the target file (before renaming, delete the target file because webhdfs doesn't support rename overwrite).